### ISSUE TITLE
Controlling Model Structure for Later Output to ONNX

### ISSUE THREAD

**Issue by JohnCraigPublic** [microsoft/CNTK#3708]
I need a little guidance on how to configure my CNTK model ( created via BrainScript ) in order that I can later get a clean model out in ONNX format.    My current issue can be seen by considering a simple model that takes a 161x161 image and does regression to a single output "z".   The section of Brainscript that defines nodes is here:
-----------------------------------------------------------------------------
        # inputs
        features = Input {imageShape}
        featNorm = features - Constant(114)
        regr = Input {labelDim}

        # apply model to features
        z = model (featNorm)

        # loss and error computation
        diff = regr - z
        sqerr = ReduceSum (diff.*diff, axis=1)

        # declare special nodes
        featureNodes    = (features)
        labelNodes      = (regr)
        criterionNodes  = (sqerr)
        evaluationNodes = (sqerr)
        outputNodes     = (z)
--------------------------------------------------------------------------

I believe that to be totally standard sort of Brainscript - -and the resulting model works great
when I call it from C++ in my inference engine.   

But when I output it as ONNX and try to inference it, there are some difficulties.  But stepping back, lets look at my CNTK model using Netron.  The top input of the model and everything looks good until we get down to the output stage.  Attached is a screen capture from Netron.
![OutputOfModel](https://user-images.githubusercontent.com/14355377/60541682-02264c00-9cc7-11e9-8a4d-7493a8d93db6.jpg)

The problem is that is shows an input called "regr" which is something used at train time, but something I don't want in the run-time model to be used when inferencing.   How can I get a "clean" model with one input (the image) and one output (z) ?   I must be missing something...

**Comment by delzac** [comment]
How do you save your model in BrainScript. Typically in python it goes something like this:

    model = Model()
    prediction = model(input_image_tensor)
    loss = loss_fucntion(prediction, truth)
    
    prediction.save(FILE_PATH,)  # this saved the model without the train-time truth tensor appearing
    loss.save(FILEPATH, )  # you will get the train-time truth tensor appearing.

**Comment by JohnCraigPublic** [comment]
I'm not sure.  It seems to happen automatically "by magic" with BrainScript.

**Comment by delzac** [comment]
Then with whatever software you are using to work on the onnx model, can you select the `z` node in the graph? Computing the `z` node should not require the train-time tensor.
